src/js/view/components/LogViewer.js

```diff
@@ -113,34 +121,67 @@ const LogViewer = props => {
     };
   }, [
     props.totalNrOfLinesForFiles[props.source.path],
-    props.indexesForNewLines[props.source.path]
+    props.indexesForNewLines[props.source.path],
+    props.totalNrOfFilteredLines[props.source.path]
   ]);
 
   useEffect(() => {
     /* Effect for when a new filter or highlight is applied,
     send the lines to be filtered and highlighted again */
-    if (props.logs[props.source.path]) {
+    let logs;
+    filterRef.current = filterInput;
+
+    if (filterInput.length !== 0) {
+      fetchFilteredLinesFromBackend(
+        props.source.path,
+        parseRegExp(filterInput).toString()
+      );
+      logs = props.filteredLogs[props.source.path];
+    } else {
+      logs = props.logs[props.source.path];
+    }
+
+    if (logs) {
       // Reset the previous lines count, as all lines should be wiped.
+      previousFilteredLinesLength.current = 0;
       previousLinesLength.current = 0;
       sendMessageToHiddenWindow({
-        logs: props.logs[props.source.path]
+        logs
       });
     }
-  }, [filterInput, highlightInput, highlightColor]);
+  }, [filterInput, highlightInput, highlightColor, filterRef.current]);
 
   useEffect(() => {
     // Effect for when new lines are added
-    if (props.logs[props.source.path]) {
+    if (props.logs[props.source.path] && filterInput.length === 0) {
       /* Only send lines one by one if there already are lines set.
       Slice used so only newer lines is sent or the entire array if no lines */
       let newLines = props.logs[props.source.path].slice(
         previousLinesLength.current
       );
+
       sendMessageToHiddenWindow({
         sendLinesOneByOne: previousLinesLength.current > 0 ? true : false,
         logs: newLines
       });
       previousLinesLength.current = props.logs.length;
+    } else if (filterInput.length !== 0) {
+      fetchFilteredLinesFromBackend(
+        props.source.path,
+        parseRegExp(filterInput).toString()
+      );
```

Comment:
This causes it to fetch for filtered lines everytime a new log has been added if the filterinput is not empty. I guess this maybe could a bit taxing, and if you believe so this could be removed for the time being. If we do that we won't be able to filter files live without having to rewrite the filterinput query.